sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java

```diff
@@ -94,6 +104,36 @@
    */
   Table loadTable(Identifier ident) throws NoSuchTableException;
 
+  /**
+   * Load table metadata by {@link Identifier identifier} from the catalog.
+   * <p>
+   * If the catalog supports views and contains a view for the identifier and not a table, this
+   * must throw {@link NoSuchTableException}.
+   *
+   * @param ident a table identifier
+   * @param version version of the table
+   * @return the table's metadata
+   * @throws NoSuchTableException If the table doesn't exist or is a view
+   */
+  default Table loadTable(Identifier ident, String version) throws NoSuchTableException {
```

Comment:
should we define the `version` parameter type as `int`?

Reply:
I don't think so. Version is specific to a data source, so passing it as a string is a good idea.

Reply:
@rdblue - got it, I agree with it. It's not consistent with version definition in parser `SqlBase.g4`, so asked here.